qiita_pet/handlers/api_proxy/util.py

```diff
@@ -37,3 +40,31 @@ def check_access(study_id, user_id):
         return {'status': 'error',
                 'message': 'User does not have access to study'}
     return {}
+
+
+def check_fp(study_id, filename):
+    """Check whether an uploaded file exists
+
+    Parameters
+    ----------
+    study_id : int
+        Study file uploaded to
+    filename : str
+        name of the uploaded file
+
+    Returns
+    -------
+    dict or str
```

Comment:
I suppose, although this means the contents of the `file` key will be different for error and non-error, since the full filepath will be there on success and just the filename on error.